woocommerce/woocommerce-ios
# Blaze Optimization: Navigation from the section on My Store

Closes: #10951 


## Description

This PR adds navigation to the Blaze section on the My Store screen. Changes include:
- Added new sources for the Blaze entry point.
- Intro view: replace the `NavigationView` with a `VStack` to get rid of the divider under the navigation bar.
- `BlazeCampaignDashboardViewHostingController`:
  - Injected a parent navigation controller to the initializer. This is needed to navigate to other screens.
  - Handled navigation to the campaign creation flow and the campaign list.
- `BlazeCampaignDashboardView` and its model:
  - Check if there is any existing campaign to show the intro view upon selecting campaign creation.
  - Handle navigation to campaign detail.

## Testing instructions

- Log in as an admin to a site that is eligible for Blaze but doesn't have any campaign yet.
- If the site has any existing product, the My Store screen should show the Blaze section with the latest product. If not, create one.
- Tap the Create campaign button. The intro screen should be displayed.
- Tap the Start campaign button on the intro screen, the campaign creation flow should start. Notice in Xcode console: `🔵 Tracked blaze_flow_started, properties: [AnyHashable("source"): "intro_view", ...]`
- Navigate back, tap the product item > tap Start campaign on the intro view. A web view for campaign creation should be displayed with details for that product right away.
- Create a campaign - upon completion, the web view should be dismissed and the campaign list should be displayed.
- Notice that a bottom sheet is displayed saying the new campaign has been submitted for review.
- Tap Back, the My Store screen should displayed the new campaign in the Blaze section now.
- Tap the campaign, the detail screen for the campaign should be displayed.
- Dismiss the detail screen and tap the Create campaign button. The campaign creation flow should be displayed immediately without the intro screen. Notice in Xcode console: `🔵 Tracked blaze_flow_started, properties: [AnyHashable("source"): "my_store_section_create_campaign_button", ...]`
- Navigate back, and tap the Show all campaigns button. The campaign list should be displayed without the tip about any created campaign.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/58da902f-b8ea-45b0-8f9d-7a4ee12beb8a



---
